pennylane/gradients/vjp.py

```diff
@@ -370,6 +370,7 @@ def func(_, num=None):  # pylint: disable=unused-argument
                     multi = len(tape.measurements) > 1
                     if multi:
                         multi_dy = dy[0]
+                        res = qml.math.convert_like(res, multi_dy)

```

Comment:
Was this required to preserve the Torch device used?

Reply:
Yes right!